File: python/svcomp_create_slave.py
import random
import pprint
	
class slave_item:
	def __init__(self, nxt = None, prev = None):
		self.next = nxt
		self.prev = prev

def insert_slave(x):
	item = slave_item(None, None)
	nxt = x
	item.next = nxt
	if nxt is not None:
		nxt.prev = item
	return item

def create_slave(n):
	dll = None
	while n > 0:
		n = n - 1
		dll = insert_slave(dll)
	return dll

def main():
	create_slave(5)

if __name__=="__main__":
	main()

File: java/retrieve heap/src/testsuites/SVcomp_Create_Slave.java
package testsuites;

public class SVcomp_Create_Slave {
    static Slave dll_create_slave(int n)
    {
        //pre
        Slave dll = null;
        //dll = dll_insert_slave(dll);
        //dll = dll_insert_slave(dll);
        while(n > 0)
        {
            //loop
            n--;
            dll = insert_slave(dll);
        }
        //post
        return dll;
    }

    static Slave insert_slave(Slave x)
    {
        Slave item = new Slave();

        item.next = null;
        item.prev = null;

        Slave next = x;
        item.next = next;
        if (next != null) {
            next.prev = item;
        }
        return item;
    }

    public static void main(String args[]){
        dll_create_slave(10);
    }

}

class Slave{
    Slave next;
    Slave prev;
}
